src/ast/mod.rs

```diff
@@ -4376,7 +4410,31 @@ impl fmt::Display for Statement {
                     full = if *full { "FULL " } else { "" },
                 )?;
                 if let Some(db_name) = db_name {
-                    write!(f, " FROM {db_name}")?;
+                    let keyword = match &db_name_keyword {
+                        Some(Keyword::FROM) => "FROM",
+                        Some(Keyword::IN) => "IN",
+                        _ => unreachable!(),
+                    };
+                    write!(f, " {} {db_name}", keyword)?;
+                }
```

Comment:
I think representation wise we can represent the IN/FROM explicitly? e.g.

```rust
enum ShowClause {
   IN,
   FROM
}
Statement::ShowViews{ clause: Option<ShowClause>, db_name: Option<Ident> }
```
The unreachable panic I don't think fits well here since there's no invariant guarding that panic (the parser codepath is quite far away from this and could possibly change subtly to trigger it), I think it'd be reasonable as in the previous version to display an incorrect sql (which would be less consequential given we have a bug vs than crash the user's service). Though in this case likely we can rely on the compiler to make such a state unavoidable via the enum route.

Also would it make more sense to skip the nested `if let` since the db_name and db_name_keyword are set independently. as in

```rust
if let Some(db_name_keyword) = db_name_keyword {
    write(f, " {db_name_keyword}")?;
}
if let Some(db_name) = db_name {
    write(f, " {db_name})?;
}
```